tests/test_mutations.py

```diff
@@ -373,6 +373,53 @@ def test_input_update_mutation(db, gql_client: GraphQLTestClient):
     assert set(issue.tags.all()) == set(expected_tags)
 
 
+@pytest.mark.django_db(transaction=True)
+def test_input_update_mutation_m2m_not_nullable(db, gql_client: GraphQLTestClient):
+    query = """
+    mutation UpdateProject ($input: ProjectInputPartial!) {
+      updateProject (input: $input) {
+        __typename
+        ... on OperationInfo {
+          messages {
+            kind
+            field
+            message
+          }
+        }
+        ... on ProjectType {
+          id
+          name
+          status
+          dueDate
+          milestones {
+            id
+            name
+          }
+          cost
+        }
+      }
+    }
+    """
+    project = ProjectFactory.create(
+        name="Project Name",
+    )
+    milestone_1 = MilestoneFactory.create(project=project)
+    MilestoneFactory.create(project=project)
+
+    res = gql_client.query(
+        query,
+        {
+            "input": {
+                "id": to_base64("ProjectType", project.pk),
+                "milestones": [{"id": to_base64("MilestoneType", milestone_1.pk)}],
+            }
+        },
+    )
+    assert res.data and isinstance(res.data["updateProject"], dict)
+
+    assert len(res.data["updateProject"]["milestones"]) == 1
```

Comment:
It would be nice to check that the only remaining milestone is really the milestone_1.

Reply:
Yes, sure, makes sense. I added an assertion for this.